### PR TITLE
test: Add dashboard content check and search epochs and blocks tests

### PR DESCRIPTION
**Pull request by Luisdzc**
## Description

This pr contains new automation test for dashboard feature, like basic content validation and couple of test for the search logic from dashboard, checking proper search redirection and data validation using BlockFrost services

### Issue ticket number and link

- [x] This PR has a valid ticket number or issue: [[link](https://cardanofoundation.atlassian.net/browse/MET-2154)]

### Testing & Validation

- [x] This PR has been tested/validated in Chrome, Firefox, Safari and Brave browsers.
- [x] The code has been tested locally with test coverage match expectations.
- [] Added new Unit/Component testing (if relevant).

### Security

- [x] No secrets are being committed (i.e. credentials, PII)
- [x] This PR does not have any significant security implications

### Code Review

- [x] There is no unused functionality or blocks of commented out code (otherwise, please explain below)
- [x] In addition to this PR, all relevant documentation (e.g. Confluence / README.md file) and architecture diagrams (e.g. Miro) were updated

### Design Review

- [ ] If this PR contains changes to the UI, it has gone through a design review with UX Designer or Product owner.
- [ ] In case PR contains changes to the UI, add some screenshots to notice the differences
---
##### _Before_

[comment]: <> (Add screenshots)

##### _After_

[comment]: <> (Add screenshots)

##### _Before_

[comment]: <> (Add screenshots)

##### _After_

[comment]: <> (Add screenshots)

##### _Before_

[comment]: <> (Add screenshots)

##### _After_

[comment]: <> (Add screenshots)

### CODE REVIEW

File: playwright/helpers/endpoints.ts
Comment (obstar) [inline playwright/helpers/endpoints.ts:1]: i would refactor some function pairs e.g. GetBlockByNumber and LastBlockData they have common base "/blocks" same for LastEpochData and LastEpochData "/epochs"

File: playwright/pages/block-detail.page.ts [hunk 1]
```diff
@@ -12,16 +12,36 @@ export function blockDetailPage(page: Page) {
   const blockNumber = page.locator('//div[div[div[text()="Block"]]]/following-sibling::div');
   const blockTotalOutput = page.locator('//div[div[div[text()="Total output in ADA"]]]/following-sibling::div');
```
Comment (obstar) [inline playwright/pages/block-detail.page.ts:13]: add one empty line between 13 and 14 line

File: playwright/tests/e2e/steps/dashboard.steps.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { createBdd } from "playwright-bdd";
+
```
Comment (obstar) [inline playwright/tests/e2e/steps/dashboard.steps.ts:2]: remove line 2

**Review by obstar — CHANGES_REQUESTED** [review]

File: playwright/helpers/endpoints.ts [hunk 3]
```diff
@@ -13,14 +13,14 @@ export class CardanoFoundation {
 export class BlockFrost {
   static readonly BASE_MAIN_NET_URL = process.env.BLOCKFROST_API_URL;
 
-  static LastBlockData = class {
+  static BLOCKS = class {
```
Comment (obstar) [inline playwright/helpers/endpoints.ts:16]: BLOCKS -> Blocks

File: playwright/helpers/endpoints.ts [hunk 4]
```diff
@@ -13,14 +13,14 @@ export class CardanoFoundation {
...
     public static get Base() {
-      return `${BlockFrost.BASE_MAIN_NET_URL}/blocks/latest`;
+      return `${BlockFrost.BASE_MAIN_NET_URL}/blocks`;
     }
   };
-  static LastEpochData = class {
+  static EPOCHS = class {
```
Comment (obstar) [inline playwright/helpers/endpoints.ts:21]: EPOCHS -> Epochs

File: playwright/helpers/endpoints.ts [hunk 5]
```diff
@@ -13,14 +13,14 @@ export class CardanoFoundation {
 export class BlockFrost {
   static readonly BASE_MAIN_NET_URL = process.env.BLOCKFROST_API_URL;
 
-  static LastBlockData = class {
+  static BLOCKS = class {
     public static get Base() {
-      return `${BlockFrost.BASE_MAIN_NET_URL}/blocks/latest`;
+      return `${BlockFrost.BASE_MAIN_NET_URL}/blocks`;
     }
   };
```
Comment (obstar) [inline playwright/helpers/endpoints.ts:20]: public static get Latest() {
				return `${CardanoFoundation.BlockFrost.Blocks.Base}/latest`;
			}

File: playwright/helpers/endpoints.ts [hunk 6]
```diff
@@ -13,14 +13,14 @@ export class CardanoFoundation {
...
-  static LastEpochData = class {
+  static EPOCHS = class {
     public static get Base() {
-      return `${BlockFrost.BASE_MAIN_NET_URL}/epochs/latest`;
+      return `${BlockFrost.BASE_MAIN_NET_URL}/epochs`;
     }
   };
```
Comment (obstar) [inline playwright/helpers/endpoints.ts:25]: public static get Latest() {
				return `${CardanoFoundation.BlockFrost.Epochs.Base}/latest`;
			}

File: playwright/api/call-blockfrost/blockfrost.api.ts [hunk 7]
```diff
@@ -24,7 +24,37 @@ export function blockfrostApi(request: APIRequestContext) {
   const getLastEpochData = async () => {
     return BaseApi.getData(
       request,
-      Endpoint.BlockFrost.LastEpochData.Base,
+      Endpoint.BlockFrost.EPOCHS.Base + "/latest",
```
Comment (obstar) [inline playwright/api/call-blockfrost/blockfrost.api.ts:27]: /latest should be in endpoints.ts - why? becasue you want central place to handle endpoints as per comment in endpints.ts

File: playwright/api/call-blockfrost/blockfrost.api.ts [hunk 8]
```diff
@@ -9,7 +9,7 @@ export function blockfrostApi(request: APIRequestContext) {
   const getLatestBlockData = async () => {
     return BaseApi.getData(
       request,
-      Endpoint.BlockFrost.LastBlockData.Base,
+      Endpoint.BlockFrost.BLOCKS.Base + "/latest",
```
Comment (obstar) [inline playwright/api/call-blockfrost/blockfrost.api.ts:12]: /latest should be in endpoints.ts - why? becasue you want central place to handle endpoints as per comment in endpints.ts

**Review by obstar — CHANGES_REQUESTED** [review]

File: playwright/tests/e2e/features/dashboard.feature [hunk 9]
```diff
@@ -0,0 +1,25 @@
+Feature: Check explorer dashboard shows a general and updated data of the cardano blockchain
+
+  Scenario: Check Explorer dashboard have all the elements with a general resume about explorer content
+    Given User goes to explorer portal
+    Then the user should see tha general dashboard page with all the resume information
```
Comment (cfuisat) [inline playwright/tests/e2e/features/dashboard.feature:5]: tha -> the ( spelling mistake ) , what is resume information means?

File: playwright/tests/e2e/steps/dashboard.steps.ts [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import { BlockfrostEpochInformationDto } from "../../../api/dtos/blockfrostEpochInformation.dto";
+import { epochDetailPage } from "../../../pages/epoch-detail.page";
+import { BlockInformationDto } from "../../../api/dtos/blockInformation.dto";
+import { blockDetailPage } from "../../../pages/block-detail.page";
+
+const { Given, When, Then } = createBdd();
+let epochData: Promise<BlockfrostEpochInformationDto>;
+let blockData: Promise<BlockInformationDto>;
+
+Then(/^the user should see tha general dashboard page with all the resume information$/, async ({ page }) => {
```
Comment (cfuisat) [inline playwright/tests/e2e/steps/dashboard.steps.ts:15]: tha -> the ( spelling mistake )

**Review by cfuisat — APPROVED** [review]

**Pull request merged**